components/webcompat_reporter/browser/webcompat_report_uploader.h

```diff
@@ -24,19 +24,23 @@ namespace webcompat_reporter {
 
 struct Report {
   Report();
+  Report(const Report&);
+  Report& operator=(const Report&);
   ~Report();
 
-  std::string channel;
-  GURL report_url;
-  bool shields_enabled;
-  std::string ad_block_setting;
-  std::string fp_block_setting;
-  std::string ad_block_list_names;
-  std::string languages;
-  bool language_farbling;
-  bool brave_vpn_connected;
-  base::Value details;
-  base::Value contact;
+  std::optional<std::string> brave_version;
+  std::optional<std::string> channel;
+  std::optional<GURL> report_url;
+  std::optional<std::string> shields_enabled;
+  std::optional<std::string> ad_block_setting;
+  std::optional<std::string> fp_block_setting;
+  std::optional<std::string> ad_block_list_names;
+  std::optional<std::string> languages;
+  std::optional<std::string> language_farbling;
+  std::optional<std::string> brave_vpn_connected;
+  std::optional<std::string> details;
+  std::optional<std::string> contact;
+  std::optional<base::Value> ad_block_components;
```

Comment:
you added a mojo structure that's just similar to this one. Do we really need to have both? Why not just use mojo structure everywhere from now on? Seems like the right thing to do.